edi_notification_oca/models/edi_exchange_type.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2024
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class EDIExchangeType(models.Model):
+    _inherit = "edi.exchange.type"
+
+    send_notification = fields.Boolean(
+        help="The notifications will be automatically sent to all selected users"
+        " after processing the exchange record. If active, please select"
+        " the specific groups and specific users in the 'Notifications' page.",
+        default=False,
+    )
+    notification_groups_ids = fields.Many2many(
+        comodel_name="res.groups", string="Notification Groups"
+    )
+    notification_users_ids = fields.Many2many(
+        comodel_name="res.users",
+        string="Notification Users",
+        help="Select users to send notifications. If 'Notification Groups' "
+        "have been selected, notifications will also send to users selected in here.",
+    )
+
+    @api.onchange("send_notification")
+    def _onchange_send_notification(self):
+        if (
+            not self.send_notification
+            or self._origin.send_notification != self.send_notification
+        ):
+            self.notification_groups_ids = None
+            self.notification_users_ids = None
+
+    @api.constrains(
+        "send_notification", "notification_groups_ids", "notification_users_ids"
+    )
+    def _check_notification_groups_users(self):
+        for rec in self:
+            if not rec.send_notification and (
+                rec.notification_groups_ids or rec.notification_users_ids
+            ):
+                raise UserError(
```

Comment:
why not just hide notification_groups_ids and rec.notification_users_ids when send_notification is not True?